File: app/src/main/java/com/icerrate/popularmovies/utils/ViewUtils.java
package com.icerrate.popularmovies.utils;

import android.app.Activity;
import android.app.ProgressDialog;
import android.content.Context;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AlertDialog;
import android.view.Display;
import android.view.Surface;
import android.view.View;
import android.view.WindowManager;
import android.view.inputmethod.InputMethodManager;

import com.icerrate.popularmovies.R;

/**
 * @author Ivan Cerrate.
 */

public class ViewUtils {

    private final static int DEFAULT_COLUMNS = 2;

    private final static int HORIZONTAL_COLUMNS = 4;

    public static Snackbar createSnackbar(View parent, String message, int duration) {
        return Snackbar.make(parent, message, duration)
                .setActionTextColor(parent.getResources().getColor(R.color.colorPrimary));
    }

    public static Snackbar createSnackbar(View parent, int resId, int duration) {
        return Snackbar.make(parent, resId, duration)
                .setActionTextColor(parent.getResources().getColor(R.color.colorPrimary));
    }

    public static int getGridColumns(Activity activity) {
        if (activity != null && activity.getWindowManager() != null) {
            Display display = activity.getWindowManager().getDefaultDisplay();
            if (display.getRotation() == Surface.ROTATION_0) {
                return DEFAULT_COLUMNS;
            } else {
                return HORIZONTAL_COLUMNS;
            }
        }
        return DEFAULT_COLUMNS;
    }
}

File: app/src/main/java/com/icerrate/popularmovies/utils/ViewUtils.kt
package com.icerrate.popularmovies.utils

import android.app.Activity
import androidx.core.content.ContextCompat
import com.google.android.material.snackbar.Snackbar
import android.view.Display
import android.view.Surface
import android.view.View
import android.view.WindowManager
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowCompat
import android.os.Build
import com.icerrate.popularmovies.R

/**
 * @author Ivan Cerrate.
 */
object ViewUtils {

    private const val DEFAULT_COLUMNS = 2
    private const val HORIZONTAL_COLUMNS = 4

    fun createSnackbar(parent: View, message: String, duration: Int): Snackbar {
        return Snackbar.make(parent, message, duration)
            .setActionTextColor(ContextCompat.getColor(parent.context, R.color.colorPrimary))
    }

    fun createSnackbar(parent: View, resId: Int, duration: Int): Snackbar {
        return Snackbar.make(parent, resId, duration)
            .setActionTextColor(ContextCompat.getColor(parent.context, R.color.colorPrimary))
    }

    fun getGridColumns(activity: Activity?): Int {
        return if (activity?.windowManager != null) {
            val rotation = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
                activity.display?.rotation ?: Surface.ROTATION_0
            } else {
                @Suppress("DEPRECATION")
                activity.windowManager.defaultDisplay.rotation
            }
            if (rotation == Surface.ROTATION_0) {
                DEFAULT_COLUMNS
            } else {
                HORIZONTAL_COLUMNS
            }
        } else {
            DEFAULT_COLUMNS
        }
    }
}